entity_test.go

```diff
@@ -53,6 +53,26 @@ func (g *getComponentSystem) UpdateEntity(entity *Entity, dt float32) {
 	}
 }
 
+func TestDuplication(t *testing.T) {
+	e := NewEntity([]string{})
+	amount := 20
+
+	es := e.Duplicate(amount)
+
+	if amount != len(es) {
+		t.Errorf("Not same amount of entities %v %v", amount, len(es))
+	}
+}
+
+func TestClone(t *testing.T) {
+	e := NewEntity([]string{})
+	e2 := e.Clone()
+
+	if e.ID() == e2.ID() {
+		t.Error("IDs are the same :/ (a statistical anomaly may have occurred.)")
```

Comment:
I think the current ID system is OK :tm: 

When it becomes a problem, we should deal with it.